source/client/options_impl.h

```diff
@@ -27,28 +27,44 @@ class OptionsImpl : public Options {
   std::string verbosity() const override { return verbosity_; };
   std::string outputFormat() const override { return output_format_; };
   bool prefetchConnections() const override { return prefetch_connections_; }
-  uint64_t burstSize() const override { return burst_size_; }
+  uint32_t burstSize() const override { return burst_size_; }
   std::string addressFamily() const override { return address_family_; };
   std::string requestMethod() const override { return request_method_; };
   std::vector<std::string> requestHeaders() const override { return request_headers_; };
   uint32_t requestBodySize() const override { return request_body_size_; };
+  const envoy::api::v2::auth::UpstreamTlsContext& tlsContext() const override {
+    return tls_context_;
+  };
+  uint32_t maxPendingRequests() const override { return max_pending_requests_; }
+  uint32_t maxActiveRequests() const override { return max_active_requests_; }
+  uint32_t maxRequestsPerConnection() const override { return max_requests_per_connection_; }
 
 private:
-  uint64_t requests_per_second_;
-  uint64_t connections_;
-  uint64_t duration_;
-  uint64_t timeout_;
+  void setNonTrivialDefaults();
+  void validate() const;
+
+  // We cap on negative values. TCLAP accepts negative values which we will get here as very
+  // large values. We just cap values, hoping we catch accidental wraparound to a reasonable extent.
+  static constexpr uint32_t largest_acceptable_uint32_option_value = UINT32_MAX - 30000;
```

Comment:
Hmm, if TCLAP accepts negative values, I wonder if that means it fails to accept positive values larger than `UINT16_MAX` for `uint32_t` args...

Reply:
Added ParserIntRangeTest, which passes, I think we're good here?